platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -174,6 +174,209 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
 
+[[modularize_builds]]
+== Modularize Your Builds
+:keywords: structuring-builds
+
+Modularize your builds by splitting your code into multiple projects.
+
+=== Explanation
+
+Splitting your build's source into multiple Gradle projects (modules) is essential for leveraging Gradle's automatic work avoidance and parallelization features.
+When a source file changes, Gradle only recompiles the affected projects.
+If all your sources reside in a single project, Gradle can't avoid recompilation and won't be able to run tasks in parallel.
+Splitting your source into multiple projects can provide additional performance benefits by minimizing each subproject's compilation classpath and ensuring code generating tools such as annotation and symbol processors run only on the relevant files.
```

Comment:
❔ Should we mention somewhere that Gradle does well with a bigger amount of source files per module due to incremental compilation?

Reply:
I also think it's more important to strongly emphasize that you should aim to benefit from work avoidance here.  This does mention incremental compilation later when explaining the example.